.github/workflows/pages.yml

```
@@ -0,0 +1,68 @@
+name: Documentation
+
+on: [push, pull_request]
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:latest
+
+    steps:
+      - name: Install dependencies
+        run: |
+          dnf install -y \
+            gcc \
+            gi-docgen \
+            git \
+            gobject-introspection-devel \
+            gtk3-devel \
+            gtk4-devel \
+            meson \
+            python3-pytest \
+            python3-dbusmock \
+            qt5-qtbase-devel \
+            qt5-qtx11extras-devel \
+            qt6-qtbase-devel \
+            qt6-qtbase-private-devel \
+            vala
+
+      - name: Check out libportal
+        uses: actions/checkout@v3
+
+      - name: Configure libportal
+        run: meson setup --prefix=/usr builddir -Dbackend-gtk3=enabled -Dbackend-gtk4=enabled -Dbackend-qt5=enabled -Dbackend-qt6=enabled
+
+      - name: Build libportal
+        run: ninja -C builddir
```

```suggestion
        run: ninja -C builddir doc/libportal-1
```

Only build what is needed.